### PR TITLE
Iss659 

### PR DESCRIPTION
**Pull request by pbutti**
Details in 
https://github.com/JeffersonLab/hps-java/issues/659

### CODE REVIEW

File: analysis/src/main/java/org/hps/analysis/MC/TrackToMCParticleRelationsDriver.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        List<LCRelation> trackToTruthTrackRelations    =  new ArrayList<LCRelation>();
+        List<Track>      truthTrackCollection          =  new ArrayList<Track>();
+                
+        for (Track track : trackCollection) {
+
+            if (debug) {
+                        System.out.println("-----PRINTING TRACK------- " + trackCollectionName);
+                        System.out.println(trackCollectionName+" Track:");
+                        System.out.printf("d0 %f z0 %f R %f phi %f lambda %s\n", trk_htf.dca(), trk_htf.z0(), trk_htf.R(), trk_htf.phi0(), trk_htf.slope());
+            }
```
Comment (mholtrop) [inline analysis/src/main/java/org/hps/analysis/MC/TrackToMCParticleRelationsDriver.java:126]: The three lines above give a "check style violation" for incorrect indentation.
Note a big deal, but it does prevent the tests from running.

File: analysis/src/main/java/org/hps/analysis/MC/TrackToMCParticleRelationsDriver.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        //Truth Tracks and Relations
+        List<LCRelation> trackToTruthTrackRelations    =  new ArrayList<LCRelation>();
+        List<Track>      truthTrackCollection          =  new ArrayList<Track>();
+                
+        for (Track track : trackCollection) {
+
+            if (debug) {
+                        System.out.println("-----PRINTING TRACK------- " + trackCollectionName);
+                        System.out.println(trackCollectionName+" Track:");
+                        System.out.printf("d0 %f z0 %f R %f phi %f lambda %s\n", trk_htf.dca(), trk_htf.z0(), trk_htf.R(), trk_htf.phi0(), trk_htf.slope());
```
Comment (mholtrop) [inline analysis/src/main/java/org/hps/analysis/MC/TrackToMCParticleRelationsDriver.java:125]: try_htf does not seem to have been defined yet at this point.

**Review by mholtrop — CHANGES_REQUESTED** [review]
See other comments.
Please make sure that the code in this issue is actually the code that you have been testing. It currently does not seem to be correct, so there may be other files that are different between what is in the repo for this issue and what you have been working with.

**Comment by pbutti** [comment]
Thanks @mholtrop ,
for checking this. Uhm.. Weird that there are these errors as locally seems to work fine. I ported additional fixes from Robert, so I'll open a new merge request with the updated branch and check these issues.

**Pull request closed without merging**